src/file/imp/unix.rs

```diff
@@ -4,15 +4,15 @@ use std::ffi::CString;
 use std::fs::{self, File, OpenOptions};
 use std::io;
 use std::os::unix::ffi::OsStrExt;
-use std::os::unix::io::{AsRawFd, FromRawFd, RawFd};
+use std::os::unix::io::FromRawFd;
 use std::path::Path;
 use util;
 
 #[cfg(all(lfs_support, target_os = "linux"))]
 use libc::{fstat64 as fstat, open64 as open, stat64 as stat_t};
 
 #[cfg(not(any(all(lfs_support, target_os = "linux"), target_os = "redox")))]
-use libc::{fstat, open, stat as stat_t};
```

Comment:
I think this is broken for redox. I believe these two lines should be changed to

```rs
#[cfg(any(target_os = "linux", target_os = "emscripten", target_os ="l4re"))]
use libc::open64 as open;

#[cfg(not(any(target_os = "linux", target_os = "emscripten", target_os ="l4re")))]
use libc::open;
```

And then, I'd:

1. Move the `stat` definition for redox into the `same_dev_ino` definition (to keep everything together).
2. Import `libc::{fstat,stat as stat_t}` there.

---

This change also:

1. Removes the `lfs_support` attribute. That's an historical artifact that should be removed.
2. Adds LFS (large file support) on emscripten and l4re.